msctl

```diff
@@ -1393,6 +1393,22 @@ overviewer() {
   # Generate the map and POI.
   $OVERVIEWER_BIN --config=$SETTINGS_FILE >> $LOG_FILE 2>&1
   $OVERVIEWER_BIN --config=$SETTINGS_FILE --genpoi >> $LOG_FILE 2>&1
+
+  # Add the Google API key to the Overviewer index.html source files.
+  for f in $(find /usr/ -name "index.html" -print | grep overviewer_core/data/web_assets)
+  do
+      if [ "$(grep "$TestString" "$f")" ] ; then
+          if [ -z "$(grep "$TestString$AddString" "$f")" ] ; then
+                  if [ ! -h "$f" ] ; then
+                      sed -i "s/\?sensor=false/&\&key=$OVERVIEWER_KEY/g" "$f"
+                  fi
+          fi
+      fi
+  done
+
+  # Add the Google API key to the index.html in the maps folder. 
+  sed -i 's/\?sensor=false/&\&key='"$OVERVIEWER_KEY"'/g' "$MAPS_LOCATION/$1/index.html"
+
   # Announce the location to access the world map to players.
```

Comment:
>The problem with adding your key after the render [this is what I did in the msctl file] is that overviewer.py seems to copy the original index.html to your webroot and then perform the time consuming tile render. This means if you're waiting for overviewer.py to complete before running the sed to insert your key, and someone has gone exploring and built a backlog of tiles to render, your map may not work for a good chunk of time.

--> this happened to my server lol

> My solution is to add the key to /usr/share/pyshared/overviewer_core/data/web_assets/index.html after installing overviewer

^ So you're right in that we only need to modify the source file, the problem is I get a permission denied error when I try to do it since the user running the `msctl` script does not have permission